### PR TITLE
Add `loguru/2.1.0` recipe

### LINKED ISSUE #14542: [request] loguru/2.1.0
### Package Details
  * Package Name/Version: **loguru/2.1.0**
  * Website: **https://emilk.github.io/loguru/index.html**
  * Source code: **https://github.com/emilk/loguru**

### Description Of The Library / Tool

Loguru: a lightweight and flexible C++ logging library.

### PR DESCRIPTION
**Pull request by antekone**
Loguru: a lightweight and flexible C++ logging library.

Website: https://emilk.github.io/loguru/index.html Source code: https://github.com/emilk/loguru

Closes #14542

Specify library name and version:  **loguru/2.1.0**

Loguru is a nice lightweight logging library. It has lots of stars on GitHub, and I think it should be supported by Conan. This is my first package so I'm sorry for any oversights!

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/loguru/all/CMakeLists.txt
Comment (uilianries) [inline recipes/loguru/all/CMakeLists.txt:1]: please, submit this cmake script to the upstream first.

File: recipes/loguru/all/LICENSE.md
Comment (uilianries) [inline recipes/loguru/all/LICENSE.md:1]: License file should not be copied directly. Remove this file please

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:20]:
```suggestion
```
Version should not be defined, so the recipe can be generic to any version.

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:25]: Correct! https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.53.0"
```
It does not need 1.54.0.

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:42]:
```suggestion
        "scope_text_size": [None, "ANY"],
        "filename_width": [None, "ANY"],
        "threadname_width": [None, "ANY"],
        "scope_time_precision": [None, "ANY"],
```
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#any-special-value

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:64]:
```suggestion
```
it does not need cmake 3.25 for a very simple library.

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:74]:
```suggestion
```
Do not use `print` method in recipes, the log output is managed directly by conan client.

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:114]:
```suggestion
        copy(self, pattern='LICENSE.md', src=self.source_folder, dst=os.path.join(self.package_folder, 'licenses'))
```
extract it from the source file instead. https://docs.conan.io/en/latest/howtos/extract_licenses_from_headers.html

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution, please, take a look on my review, there are some points which should be updated.

File: recipes/loguru/all/conanfile.py
Comment (antekone) [reply recipes/loguru/all/conanfile.py:63]: Do you mean I should remove the `cmake` build requirement altogether? Or to downgrade its version?

I mean, it currently uses CMake to build itself. Can I still remove cmake build dependency?

File: recipes/loguru/all/CMakeLists.txt
Comment (antekone) [reply recipes/loguru/all/CMakeLists.txt:1]: Upstream already has a PR related to adding full CMake build: https://github.com/emilk/loguru/pull/215, but it's not merged since April.

I can use the CMake script from this un-merged PR. But in the meantime I've decided to use a minimal own CMake script only for Conan package, and migrate to the "official" CMake build once upstream's PR-215 will be merged.

**Comment by antekone** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

I want to fix issues in this PR, but I'm waiting for feedback from the Conan team.

**Review by uilianries — COMMENTED** [review]
@antekone Please, consider https://github.com/antekone/conan-center-index/pull/1

I added some improvements over your original cmake file 😄

**Comment by antekone** [comment]
@uilianries Merged your suggestions, will tidy up the latest comments from the linter bot (warnings regarding Conan v2) when I find some time for it.

**Comment by uilianries** [comment]
@antekone Thank you! Currently we are under maintenance due Conan v2 migration. Please, follow https://status.conan.io to get more information! Happy March!

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:77]:
```suggestion
        cmake_layout(self, src_folder='src')
```

File: recipes/loguru/all/conanfile.py
Comment (uilianries) [inline recipes/loguru/all/conanfile.py:77]:
```suggestion
        cmake_layout(self, src_folder='.')
```

**Comment by czoido** [comment]
Hi @antekone,
Looks like after some changes the recipe is almost there. Could you please sign the CLA so it can be merged after the last reviews?

**Comment by antekone** [comment]
@czoido CLA signed. I'm sorry for not participating in updates, and thank you for your help.

**Comment by czoido** [comment]
> @czoido CLA signed. I'm sorry for not participating in updates, and thank you for your help.

No problem! Thanks a lot for the contribution! I hope the recipe can be merged soon

**Review by czoido — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**
